lib/routes/developer/activate.js

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const auth = require('../../auth');
+const db = require('../../db');
+const hex = require('buf').to.hex;
+const AppError = require('../../error');
+
+function developerResponse(developer) {
+  return {
+    developerId: hex(developer.developerId),
+    email: developer.email,
+    createdAt: developer.createdAt
+  };
+}
+
+/*jshint camelcase: false*/
+module.exports = {
+  auth: {
+    strategy: auth.AUTH_STRATEGY,
+    scope: [auth.SCOPE_CLIENT_MANAGEMENT]
+  },
+  handler: function activateRegistration(req, reply) {
+    var email = req.auth.credentials.email;
+
+    return db.getDeveloper(email)
+      .done(
+        function(developer) {
+          if (developer) {
+            return reply(developerResponse(developer)).code(200);
+          } else {
+            return db.activateDeveloper(email)
+              .then(function() {
+                return db.getDeveloper(email);
+              })
+              .done(function(developer) {
+                // check to make sure activation succeeded
+                if (! developer) {
+                  return reply(new AppError({
```

Comment:
I'd personally restructure this one to this:

```js
return db.getDeveloper(email)
  .then(function(developer) {
    if (developer) {
      return developer;
    } else {
      // better to just return the developer here as well, instead of using an additional query
      return db.activeDeveloper(email);
    }
  })
  .then(developerResponse)
  .done(reply, reply);
```